packages/language-server/src/server.ts

```diff
@@ -99,6 +124,37 @@ documents.onDidChangeContent(async (changeEvent) => {
   }
 });
 
+function updateDocumentSettings(resource: string): void {
+  if (!hasConfigurationCapability) {
+    return;
+  }
+  if (!hasSettingsForDocument(resource)) {
+    const result = connection.workspace.getConfiguration({
+      scopeUri: resource,
+      section: "UI5LanguageAssistant",
+    });
+    setSettingsForDocument(resource, result);
+  }
+}
+
+connection.onDidChangeConfiguration((change) => {
+  if (hasConfigurationCapability) {
+    // Reset all cached document settings
+    clearSettings();
+  } else {
+    if (change.settings.UI5LanguageAssistant !== undefined) {
+      setGlobalSettings(change.settings.UI5LanguageAssistant);
+    }
+  }
+  // No further actions are required currently during configuration change. In the future we might want to
+  // re-validate the files.
+});
+
+// Only keep settings for open documents
+documents.onDidClose((e) => {
+  clearDocumentSettings(e.document.uri);
```

Comment:
we have a **per document** specific setting here?

Reply:
Yes. It's taken directly from the implementation guide. The workspace configuration request receives the document URI.
I don't think there is a way to set settings per document in vs code or theia, it's on either user or workspace level, but this is the API.